module3/homework/passwordCheck/validation.cpp

```diff
@@ -1,2 +1,69 @@
+#include <cstring>
+#include <algorithm>
+
+
 #include "validation.hpp"
-// TODO: Put implementations here
\ No newline at end of file
+
+const int8_t minPasswordLen = 9;
+std::string getErrorMessage(ErrorCode code)
+{
+    switch(code)
+    {
+        case ErrorCode::Ok : {return "OK";}
+        case ErrorCode::PasswordNeedsAtLeastNineCharacters : 
+                    {return "ERROR. Password needs at least nine characters.";}
+        case ErrorCode::PasswordNeedsAtLeastOneNumber : 
+                    {return "ERROR. Password needs at least one number.";}
+        case ErrorCode::PasswordNeedsAtLeastOneSpecialCharacter : 
+                    {return "ERROR. Password needs at least one special character.";}
+        case ErrorCode::PasswordNeedsAtLeastOneUppercaseLetter : 
+                    {return "ERROR. Password needs at least one uppercase letter.";}
+        case ErrorCode::PasswordsDoesNotMatch: 
+                    {return "ERROR. Password does not match.";}
+    }
+}
+
+bool doesPasswordsMatch(std::string password, std::string repeated_password)
+{
+    if(password != repeated_password) return false;
+    return true;
+}
+
+ErrorCode checkPasswordRules(std::string const password)
+{
+    if(password.length() < minPasswordLen) return ErrorCode::PasswordNeedsAtLeastNineCharacters;
+
+    if(std::find_if(password.begin(), 
+        password.end(), isdigit) == password.end())
+        return ErrorCode::PasswordNeedsAtLeastOneNumber;
```

Comment:
This formatting is very misleading. Please do not use it. And always use braces, even for one-line if statements.
```suggestion
    if(std::find_if(password.begin(), password.end(), isdigit) == password.end()) {
        return ErrorCode::PasswordNeedsAtLeastOneNumber;
    }
```